### PR TITLE
CLI: Plumb nonce-stored fees

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

CLI doesn't know about nonce-account-stored fee calculators

#### Summary of Changes

Plumb them through

~WIP until #8743~

### CODE REVIEW

**Comment by t-nelson** [comment]
@CriesofCarrots @jstarry this is ready for review now :pray:

File: cli/src/offline/blockhash_query.rs [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+
+#[derive(Debug, PartialEq)]
+pub enum BlockhashQuery {
+    None(Hash),
+    FeeCalculator(Source, Hash),
+    All(Source),
+}
+
+impl BlockhashQuery {
+    pub fn new(
```
Comment (CriesofCarrots) [inline cli/src/offline/blockhash_query.rs]: Fwiw, I think this constructor method is actually harder to grok in all those tests than just creating the BlockhashQuery as needed. One option would be to move this logic into `new_from_matches`, which I think is the only place you really need it, right?

Comment (t-nelson) [reply cli/src/offline/blockhash_query.rs]: Ha! `new()` was broken out of `new_from_matches()` upon request _to facilitate testing_ during review when this feature originally came in.

I _do_ like being able to test the construction logic without building an `ArgMatches`, but could go either way in the actual test case instantiations. The main motivation of using `new()` is that the `Source` instantiation is a bit verbose

Comment (CriesofCarrots) [reply cli/src/offline/blockhash_query.rs]: Lol! 🤡 🎈 feel free to disregard my comment, then

File: cli/src/offline/blockhash_query.rs [hunk 2]
```diff
@@ -0,0 +1,394 @@
+use super::*;
+
+#[derive(Debug, PartialEq)]
+pub enum Source {
+    Cluster,
+    NonceAccount(Pubkey),
+}
+
+impl Source {
+    pub fn get_blockhash_fee_calculator(
```
Comment (jstarry) [inline cli/src/offline/blockhash_query.rs]: This method name reads to me like "get the fee calculator for this blockhash" which is what the other method does.

File: cli/src/offline/blockhash_query.rs (hunk 1 above)
Comment (jstarry) [reply cli/src/offline/blockhash_query.rs]: Came here to say the same thing, the constructor approach makes it hard to tell if the tests are setup correctly. Can we switch the tests back to the enum approach?

File: cli/src/offline/blockhash_query.rs (hunk 2 above)
Comment (t-nelson) [reply cli/src/offline/blockhash_query.rs]: That's a legacy method name, but I can toss an "and" in there, sure

File: cli/src/offline/blockhash_query.rs (hunk 1 above)
Comment (t-nelson) [reply cli/src/offline/blockhash_query.rs]: Two votes wins.  First thing on the plate in the morning

File: cli/src/offline/blockhash_query.rs [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+        &self,
+        rpc_client: &RpcClient,
+    ) -> Result<(Hash, FeeCalculator), Box<dyn std::error::Error>> {
+        match self {
+            BlockhashQuery::None(hash) => Ok((*hash, FeeCalculator::default())),
+            BlockhashQuery::FeeCalculator(source, hash) => {
+                let fee_calculator = source
+                    .get_fee_calculator(rpc_client, hash)?
+                    .ok_or(format!("Has has expired {:?}", hash))?;
```
Comment (jstarry) [inline cli/src/offline/blockhash_query.rs]: nit: Has should be Hash

Comment (t-nelson) [reply cli/src/offline/blockhash_query.rs]: Good eye!

File: cli/src/offline/blockhash_query.rs (hunk 2 above)
Comment (t-nelson) [reply cli/src/offline/blockhash_query.rs]: https://github.com/solana-labs/solana/pull/8750/commits/48b274dd6dc816a79ee51d45c433d094f3afecb8

File: cli/src/offline/blockhash_query.rs (hunk 3 above)
Comment (t-nelson) [reply cli/src/offline/blockhash_query.rs]: https://github.com/solana-labs/solana/pull/8750/commits/f21c7da1483f3f19450fd946b272bac4fb3c391b

File: cli/src/offline/blockhash_query.rs (hunk 1 above)
Comment (t-nelson) [reply cli/src/offline/blockhash_query.rs]: https://github.com/solana-labs/solana/pull/8750/commits/26e0edfb6086e9bde4d733024d404a40cad82cfc

**Comment by t-nelson** [comment]
Thanks for the review, all!  Last batch should cover everything mentioned

File: cli/src/offline/blockhash_query.rs [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+            ),
+        );
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_blockhash_query_new_from_matches_without_nonce_fail() {
+        let test_commands = App::new("blockhash_query_test")
+            .arg(blockhash_arg())
+            // We can really only hit this case unless the arg requirements
```
Comment (CriesofCarrots) [inline cli/src/offline/blockhash_query.rs]:
```suggestion
            // We can really only hit this case if the arg requirements
```

File: cli/src/offline/blockhash_query.rs [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+            .get_matches_from(vec!["blockhash_query_test", "--sign-only"]);
+        BlockhashQuery::new_from_matches(&matches);
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_blockhash_query_new_from_matches_with_nonce_fail() {
+        let test_commands = App::new("blockhash_query_test")
+            .arg(blockhash_arg())
+            // We can really only hit this case unless the arg requirements
```
Comment (CriesofCarrots) [inline cli/src/offline/blockhash_query.rs]:
```suggestion
            // We can really only hit this case if the arg requirements
```

File: cli/src/offline/blockhash_query.rs (hunk 4 above)
Comment (t-nelson) [reply cli/src/offline/blockhash_query.rs]: I hate when my brain decides to rewrite a sentence before I finish typing it...

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to CI failure

**Pull request merged**